Add ChatQnA docker-compose example on AIPC using MariaDB Vector

RazvanLiviuVarzaru opened the pull request:
## Description

Add ChatQnA docker-compose example on AIPC using **MariaDB Vector**.
DepensOn: https://github.com/RazvanLiviuVarzaru/GenAIComps/pull/1

## Issues

`n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds new functionality)

## Dependencies

`n/a`

## Tests

Set the HF API token environment variable and:
```
cd ChatQnA/tests
bash test_compose_mariadb_on_aipc.sh
```

cvicentiu commented on ChatQnA/docker_compose/intel/cpu/aipc/set_env_mariadb.sh:
```diff
@@ -0,0 +1,30 @@
+export MARIADB_PASSWORD=${MARIADB_PASSWORD}
+export HUGGINGFACEHUB_API_TOKEN=${HUGGINGFACEHUB_API_TOKEN}
+export EMBEDDING_MODEL_ID="BAAI/bge-base-en-v1.5"
+export EMBEDDING_LENGHT=768 # of MODEL_ID
```
Following the langchain changes, we won't need this any more.

RazvanLiviuVarzaru replied:
Yes, now that langchain-mariadb 0.0.16 is out, EMBEDDING_LENGHT goes away.

cvicentiu commented on ChatQnA/docker_compose/intel/cpu/aipc/compose_mariadb.yaml:
```diff
@@ -0,0 +1,170 @@
+      https_proxy: ${https_proxy}
+      MARIADB_CONNECTION_STRING: mariadb+mariadbconnector://${MARIADB_USER}:${MARIADB_PASSWORD}@mariadb-server:3306/${MARIADB_DATABASE}
+      TEI_EMBEDDING_ENDPOINT: http://tei-embedding-service:80
+      EMBEDDING_LENGHT: ${EMBEDDING_LENGHT}
```
This goes away following mariadb langchain changes.

RazvanLiviuVarzaru replied:
Yes, now that langchain-mariadb 0.0.16 is out, `EMBEDDING_LENGHT` goes away.

cvicentiu commented on ChatQnA/tests/test_compose_mariadb_on_aipc.sh:
```diff
@@ -0,0 +1,207 @@
+function main() {
+
+    echo "::group::start_docker"
+    stop_docker
```
Ehm, start_docker, but we stop_docker? I think something is off here.

RazvanLiviuVarzaru replied:
An unfortunate copy-paste. It should be stop.

cvicentiu commented on ChatQnA/tests/test_compose_mariadb_on_aipc.sh:
```diff
@@ -0,0 +1,207 @@
+#!/bin/bash
+# Copyright (C) 2024 Intel Corporation
```
Copyright MariaDB Foundation, here and in all new introduced files.

cvicentiu commented on ChatQnA/docker_compose/intel/cpu/aipc/compose_mariadb.yaml:
```diff
@@ -0,0 +1,170 @@
+      retries: 3
+
+  dataprep-mariadb-vector:
+    # image: ${REGISTRY:-opea}/dataprep:${TAG:-latest}
```
Remember to change this.

cvicentiu commented on ChatQnA/docker_compose/intel/cpu/aipc/compose_mariadb.yaml:
```diff
@@ -0,0 +1,170 @@
+      https_proxy: ${https_proxy}
+    command: --model-id ${EMBEDDING_MODEL_ID} --auto-truncate
+  retriever:
+    # image: ${REGISTRY:-opea}/retriever:${TAG:-latest}
```
Remember to change this, use ENV variables to test.

cvicentiu commented on ChatQnA/docker_compose/intel/cpu/aipc/README_mariadb.md:
````diff
@@ -0,0 +1,308 @@
+```bash
+mkdir ~/OPEA -p
+cd ~/OPEA
+git clone https://github.com/opea-project/GenAIExamples.git
````
Ok, so this will work once the retriever / dataprep images for mariadb are published to Opea repo, correct? Otherwise, one needs to build the images from scratch as according to instructions below.

RazvanLiviuVarzaru replied:
Yes, either images containing the MariaDB Vector component are already published or one needs to build them and control via ENV which image to use.

Either way, when opening the Pull Request I will reference the comps PR and that should go first.

RazvanLiviuVarzaru committed:
Add ChatQnA docker-compose example on AIPC using MariaDB Vector

Example on how to deploy the ChatBot on AIPC by using MariaDB Server as a vectorstore.
- use MariaDB Server as the backend database. Minimum required version is 11.7
- use the OPEA_DATAPREP_MARIADBVECTOR component for dataprep microservice
- use the OPEA_RETRIEVER_MARIADBVECTOR component for retriever microservice

How to test
Set the HF API token environment variable and:
```
cd ChatQnA/tests
bash test_compose_mariadb_on_aipc.sh
```

Signed-off-by: Razvan-Liviu Varzaru <razvan@mariadb.org>
